psp-core.c

```diff
@@ -704,4 +704,28 @@ void PSPEmuCoreStateDump(PSPCORE hCore)
         else
             fprintf(stderr, "Querying CPU mode failed with %d\n", pspEmuCoreErrConvertFromUcErr(rcUc));
     }
+
+    /* Dump last 0x20 bytes of stack memory */
+    size_t cwLen = 8;
+    uint32_t awStack[cwLen];
+    rc = PSPEmuCoreMemRead(hCore, au32Reg[PSPCOREREG_SP] - cwLen, awStack, sizeof(awStack));
```

Comment:
The address calculation "au32Reg[PSPCOREREG_SP] - cwLen" looks wrong, it should be "au32Reg[PSPCOREREG_SP] - cwLen * sizeof(uint32_t)" most likely. I'd prefer the following for reading the stack:

uint32_t au32Stack[8];
rc = PSPEmuCoreMemRead(hCore, au32Reg[PSPCOREREG_SP] - sizeof(au32Stack), &au32Stack[0], sizeof(au32Stack));
if (!rc)
...